### PR TITLE
3957-data-store-get-all

### PR DESCRIPTION
**Pull request by alysonturing**
New Action: 
- Get all records

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3974"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/data_stores/actions/get-all-records/get-all-records.mjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    },
+  },
+  async run({ $ }) {
+    const { dataStore } = this;
+    const keys = await dataStore.keys();
+    const data = await keys.reduce(async (acc, key) => {
+      acc[key] = await dataStore.get(key);
+      return acc;
+    }, {});
+    $.export("$summary", `Successfully found ${keys.length} keys.`);
```
Comment (andrewjschuang) [inline components/data_stores/actions/get-all-records/get-all-records.mjs:25]: I think **returned** is a better word here

```suggestion
    $.export("$summary", `Successfully returned ${keys.length} keys.`);
```

File: components/data_stores/actions/get-all-records/get-all-records.mjs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      ],
+    },
+  },
+  async run({ $ }) {
+    const { dataStore } = this;
+    const keys = await dataStore.keys();
+    const data = await keys.reduce(async (acc, key) => {
+      acc[key] = await dataStore.get(key);
+      return acc;
+    }, {});
```
Comment (andrewjschuang) [inline components/data_stores/actions/get-all-records/get-all-records.mjs:24]: Wondering here, would it be better to make these `get`s async?
Using `Promise.all()` or something similar?

**Review by andrewjschuang — COMMENTED** [review]
@alysonturing thanks for the contribution!

I've added a couple of comments below

File: components/data_stores/actions/get-all-records/get-all-records.mjs (hunk 2 above)
Comment (alysonturing) [reply components/data_stores/actions/get-all-records/get-all-records.mjs]: changed to use promise.all

**Review by andrewjschuang — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Data_Store_3957.pdf](https://github.com/PipedreamHQ/pipedream/files/9336670/Data_Store_3957.pdf)

**Pull request merged**